pkgs/development/libraries/gtk/2.x.nix

```diff
@@ -61,6 +61,10 @@ stdenv.mkDerivation rec {
     ++ optionals cupsSupport [ cups ]
     ++ optionals stdenv.isDarwin [ AppKit Cocoa ];
 
+  preConfigure = lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11") ''
```

Comment:
to avoid mass linux rebuild
```suggestion
  preConfigure = lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11" && stdenv.isDarwin) ''
```

Reply:
Amended.

Reply:
Empty string or list don't work that way.  `null` would work, but I'm afraid such `if` would be slightly less readable.

Reply:
Should I drop it, keep as is, or do something else?

Reply:
oh right. @vcunat is right